acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java

```diff
@@ -89,13 +114,97 @@ void keyCanBeExtractedFromVaultOverTlsUsingWhitelist() throws IOException {
             30_000,
             true,
             "WHITELIST",
-            hashicorpNode.getKnownServerFilePath().get().toString(),
+            fingerprintFile.toString(),
+            null);
+
+    final String secretData = fetchSecretFromVault(configFilePath);
+
+    assertThat(secretData).isEqualTo(HashicorpVaultDocker.SECRET_VALUE);
+  }
+
+  @Test
+  void canConnectToHashicorpVaultUsingPkcs12Certificate(@TempDir final Path testDir)
+      throws IOException {
+    final String TRUST_STORE_PASSWORD = "password";
+    hashicorpNode = HashicorpNode.createAndStartHashicorp(docker, true);
+
+    final Path trustStorePath =
+        CertificateHelpers.createPkcs12TrustStore(
+            testDir, hashicorpNode.getServerCertificate().get(), TRUST_STORE_PASSWORD);
+
+    // create tomlfile
```

Comment:
nit: perhaps remove the comment about this being a toml file. I don't think we care it's a config file

Reply:
done